### PR TITLE
Use file service to listen to changes

### PR DESCRIPTION
**Pull request by sandy081**
Re #40533

Use file service to listen to changes

### CODE REVIEW

**Comment by sandy081** [comment]
@bpasero I need a method in FileChangeEvent to check if given resource is changed or not. Updated `FileChangeEvent.contains` method to take change type as optional.

File: src/vs/workbench/parts/output/electron-browser/outputServices.ts [hunk 1]
```diff
@@ -323,19 +282,26 @@ class OutputFileListener extends Disposable {
...
 		private readonly file: URI,
+		private readonly fileService: IFileService
 	) {
 		super();
 	}
 
 	watch(): void {
 		if (!this.watching) {
-			this.disposables.push(watchFile(this.file.fsPath, () => this._onDidChange.fire()));
+			this.fileService.watchFileChanges(this.file);
```
Comment (bpasero) [inline src/vs/workbench/parts/output/electron-browser/outputServices.ts]: Maybe add this to the list of `Disposable`: `toDisposable(() => this.fileService.unwatchFileChanges....)`?

**Review by bpasero — APPROVED** [review]

File: src/vs/workbench/parts/output/electron-browser/outputServices.ts (hunk 1 above)
Comment (sandy081) [reply src/vs/workbench/parts/output/electron-browser/outputServices.ts]: @bpasero `unwatch` is called in `dispose` method.

Comment (bpasero) [reply src/vs/workbench/parts/output/electron-browser/outputServices.ts]: @sandy081 yeah its fine as long as `watch` and `unwatch` are always called in sync.

File: src/vs/platform/files/common/files.ts [hunk 2]
```diff
@@ -284,22 +284,22 @@ export class FileChangesEvent {
...
 	 */
-	contains(resource: URI, type: FileChangeType): boolean {
+	contains(resource: URI, type?: FileChangeType): boolean {
 		if (!resource) {
 			return false;
 		}
 
 		return this._changes.some(change => {
-			if (change.type !== type) {
+			if (!isUndefinedOrNull(type) && change.type !== type) {
```
Comment (bpasero) [inline src/vs/platform/files/common/files.ts]: @sandy081 I would move this check one level up to reduce unnecessary checks in this potentially hot loop

Comment (sandy081) [reply src/vs/platform/files/common/files.ts]: Done

**Comment by sandy081** [comment]
@bpasero Updated with your suggestion. Merging it.

**Pull request merged**